dev/Pager/PrototypePager/PrototypePager.cs

```diff
@@ -111,23 +130,41 @@ protected override void OnApplyTemplate()
             if (PagerComboBox != null)
             {
                 PagerComboBox.SelectedIndex = SelectedIndex - 1;
-                PagerComboBox.SelectionChanged += (s, e) => {
-                    SelectedIndex = PagerComboBox.SelectedIndex + 1; };
+                PagerComboBox.SelectionChanged += (s, e) => { OnComboBoxSelectionChanged(); };
+            }
+            if (PagerNumberPanel != null)
+            {
+                InitializeNumberPanel();
+                PagerNumberPanel.ElementPrepared += OnElementPrepared;
+                PagerNumberPanel.ElementClearing += OnElementClearing;
             }
 
             OnPagerDisplayModeChanged();
+
             // This is for the initial page being loaded whatever page that might be.
             PageChanged?.Invoke(this, new PageChangedEventArgs(PreviousPageIndex, SelectedIndex - 1));
         }
+
+        private void InitializeNumberPanel()
+        {
+            if (NumberOfPages < MaxNumberOfElementsInRepeater)
+            {
+                PagerNumberPanel.ItemsSource = TemplateSettings.Pages;
+            }
+            else
+            {
+                PagerNumberPanel.ItemsSource = NumberPanelLeftMostState;
```

Comment:
>NumberPanelLeftMostState [](start = 47, length = 24)

Does this assume that the selected index at the time the template is applied is 0?

Reply:
Yeah it does, but I think that'll be problematic if this behavior is left alone. I've added a small fix to remove that assumption.